vassal-app/src/main/java/VASSAL/build/module/map/Flare.java

```diff
@@ -0,0 +1,481 @@
+      );
+    }
+  }
+
+  public boolean drawAboveCounters() {
+    return true;
+  }
+
+  public String encode(final Command c) {
+    if (c instanceof FlareCommand) {
+      return COMMAND_PREFIX + ((FlareCommand) c).getClickPoint().x + "," + ((FlareCommand) c).getClickPoint().y;
+    }
+    return null;
+  }
+
+  public Command decode(final String s) {
+    if (s.startsWith(COMMAND_PREFIX)) {
+      final int x = Integer.parseInt(s.substring(s.indexOf(":") + 1, s.indexOf(",")));
+      final int y = Integer.parseInt(s.substring(s.indexOf(",") + 1));
+      clickPoint = new Point(x, y);
+      return new FlareCommand(this);
+    }
+    return null;
+  }
+
+  public Class<?>[] getAllowableConfigureComponents() {
+    return new Class[0];
+  }
+
+  public void mouseClicked(final MouseEvent e) {
+  }
+
+  public void mousePressed(final MouseEvent e) {
+    if (!SwingUtils.isMainMouseButtonDown(e)) {
+      return;
+    }
+    if (FLARE_CTRL.equals(flareKey)) {
+      if (!SwingUtils.isSelectionToggle(e) || e.isAltDown() || e.isShiftDown()) {
+        return;
+      }
+    }
+    else if (FLARE_ALT.equals(flareKey)) {
+      if (!e.isAltDown() || e.isShiftDown() || SwingUtils.isSelectionToggle(e)) {
+        return;
+      }
+    }
+    else if (FLARE_ALT_SHIFT.equals(flareKey)) {
+      if (!e.isAltDown() || !e.isShiftDown() || SwingUtils.isSelectionToggle(e)) {
+        return;
+      }
+    }
+    else if (FLARE_CTRL_SHIFT.equals(flareKey)) {
+      if (!e.isShiftDown() || !SwingUtils.isSelectionToggle(e) || e.isAltDown()) {
+        return;
+      }
+    }
+    else if (FLARE_CTRL_ALT.equals(flareKey)) {
+      if (!e.isAltDown() || !SwingUtils.isSelectionToggle(e) || e.isShiftDown()) {
+        return;
+      }
+    }
+    else if (FLARE_CTRL_ALT_SHIFT.equals(flareKey)) {
+      if (!e.isAltDown() || !e.isShiftDown() || !SwingUtils.isSelectionToggle(e)) {
+        return;
+      }
+    }
+    clickPoint = new Point(e.getX(), e.getY());
+
+    final GameModule mod = GameModule.getGameModule();
+
+    Command c = new NullCommand();
+    reportFormat.setProperty(FLARE_LOCATION, map.locationName(clickPoint));
+    final Zone z = map.findZone(clickPoint);
+    reportFormat.setProperty(FLARE_ZONE, (z != null) ? z.getName() : "");
+    reportFormat.setProperty(FLARE_MAP, map.getMapName());
+    reportFormat.setProperty(GlobalOptions.PLAYER_NAME, (String) mod.getProperty(GlobalOptions.PLAYER_NAME));
+    reportFormat.setProperty(GlobalOptions.PLAYER_NAME_ALT, (String) mod.getProperty(GlobalOptions.PLAYER_NAME_ALT));
+    reportFormat.setProperty(GlobalOptions.PLAYER_SIDE, (String) mod.getProperty(GlobalOptions.PLAYER_SIDE));
+    reportFormat.setProperty(GlobalOptions.PLAYER_SIDE_ALT, (String) mod.getProperty(GlobalOptions.PLAYER_SIDE_ALT));
+    reportFormat.setProperty(GlobalOptions.PLAYER_ID,   (String) mod.getProperty(GlobalOptions.PLAYER_ID));
+    String reportText = reportFormat.getLocalizedText(propertySource);
```

Comment:
This is sufficient

    reportFormat.setProperty(FLARE_LOCATION, map.locationName(clickPoint));
    final Zone z = map.findZone(clickPoint);
    reportFormat.setProperty(FLARE_ZONE, (z != null) ? z.getName() : "");
    reportFormat.setProperty(FLARE_MAP, map.getMapName());
    String reportText = reportFormat.getLocalizedText(map);

Passing the map to getLocalizedText will satisify all of the Player Name/Id/Side options as map.getProperty() drops through to GameModule.getProperty() which satisifies the GlobalOptions properties.

Reply:
Plus you get the bonus advantage that the designer can specify any Map or Module level properties in the Message Format (using $$ notation) and these will also be satisfied by passing the map as the PropertySource.